bindings/java/src/org/sleuthkit/datamodel/AbstractAttribute.java

```diff
@@ -0,0 +1,378 @@
+/*
+ * Sleuth Kit Data Model
+ *
+ * Copyright 2021 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.util.Arrays;
+
+/**
+ * Attributes are a name-value pairs. Abstract Attribute provides the base
+ * functionality for a name value pair with type safety (analogous to a C union)
+ *
+ */
+abstract class AbstractAttribute {
+
+	private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray(); 
+
+	private BlackboardAttribute.Type attributeType;
+
+	private final int valueInt;
+	private final long valueLong;
+	private final double valueDouble;
+	private final String valueString;
+	private final byte[] valueBytes;
+
+	private SleuthkitCase sleuthkitCase;
+
+	private long attributeOwnerId;
+
+	
+	/**
+	 * Constructs an attribute with an integer value. The attribute should be
+	 * added to an appropriate artifact.
+	 *
+	 * @param attributeType The attribute type.
+	 * @param valueInt      The attribute value.
+	 *
+	 * @throws IllegalArgumentException If the value type of the specified
+	 *                                  attribute type is not
+	 *                                  TSK_BLACKBOARD_ATTRIBUTE_VALUE_TYPE.INTEGER.
+	 */
+	public AbstractAttribute(BlackboardAttribute.Type attributeType, int valueInt) throws IllegalArgumentException {
```

Comment:
We really don't need the throws clause for a unchecked exception.  I realize this is legacy and it doesn't hurt, I just noticed it.